datalad/core/local/status.py

```diff
@@ -387,12 +388,15 @@ def custom_result_renderer(res, **kwargs):  # pragma: no cover
         refds = res.get('refds', None)
         refds = refds if kwargs.get('dataset', None) is not None \
             or refds == os.getcwd() else None
-        path = res['path'] if refds is None \
-            else str(ut.Path(res['path']).relative_to(refds))
+        # Note: We have to force unicode for res['path'] because
+        # interface.utils encodes it on py2 before passing it to
+        # custom_result_renderer().
+        path = assure_unicode(res['path']) if refds is None \
+            else text_type(ut.Path(res['path']).relative_to(refds))
         type_ = res.get('type', res.get('type_src', ''))
         max_len = len('untracked')
         state = res.get('state', 'unknown')
-        ui.message('{fill}{state}: {path}{type_}'.format(
+        ui.message(u'{fill}{state}: {path}{type_}'.format(
```

Comment:
colorama isn't happy with something here: https://ci.appveyor.com/project/mih/datalad/builds/23828771/job/w6wf4myq5o6nyna6#L696

Reply:
> colorama isn't happy with something here: https://ci.appveyor.com/project/mih/datalad/builds/23828771/job/w6wf4myq5o6nyna6#L696

This is fixed by probing the system, as is done by get_most_obscure_supported_name(), so I've gone ahead and used that despite the compatibility issues with older Windows.